### PR TITLE
Extract SDK version from global.json in Versions.props

### PR DESCRIPTION
**Pull request by rainersigwald**
Instead of syncing the version in two places, treat global.json as canonical and extract the SDK version value from it in the build.

### CODE REVIEW

**Review by ladipro — APPROVED** [review]
Devil's advocate: Shouldn't there be only one copy of the version in the tree to avoid this need to sync?

**Comment by rainersigwald** [comment]
> Shouldn't there be only one copy of the version in the tree to avoid this need to sync?

Behold what horrors you have wrought.

**Review by BenVillalobos — COMMENTED** [review]
Woohoo!

After my initial excitement for this, I wonder if it's necessary? 

If global.json and Versions.props are out of sync the build will fail due to paths being off because of version numbers. I wonder if this would be better served as a custom script in our build definition that throws a warning if there's a diff between the two versions.

Thoughts?

**Comment by rainersigwald** [comment]
> I wonder if this would be better served as a custom script in our build definition that throws a warning if there's a diff between the two versions.

But that's not much better than today: a robot will come by and update global.json, breaking our build, and then we need to manually fix it. In the initial form this PR would have a robot come fix it immediately. Now it avoids the break entirely.

**Comment by BenVillalobos** [comment]
Ah I was thinking of devs updating a version and forgetting, not our usual bot PR's.

> Behold what horrors you have wrought.

I actually prefer this version. Will approve when the lights turn green.

**Review by BenVillalobos — APPROVED** [review]

**Review by ladipro — APPROVED** [review]

**Comment by Forgind** [comment]
I'll be honest—I would've slightly preferred the GitHub action, not because it's better (because it isn't) but because it would've been our first GitHub action, and that might have spawned others if it works well. Ah, well.

**Comment by rainersigwald** [comment]
> might have spawned others

I have little doubt we'll get more Actions going in the medium term.

File: eng/Versions.props [hunk 1]
```diff
@@ -31,7 +31,7 @@
   <PropertyGroup>
     <!-- DotNetCliVersion MUST match the dotnet version in global.json.
          Otherwise, this version of dotnet will not be installed and the build will error out. -->
-    <DotNetCliVersion>6.0.100-preview.3.21202.5</DotNetCliVersion>
+    <DotNetCliVersion>$([System.Text.RegularExpressions.Regex]::Match($([System.IO.File]::ReadAllText('$(MSBuildThisFileDirectory)..\global.json')), '"dotnet": "([^"]*)"').Groups.get_Item(1))</DotNetCliVersion>
```
Comment (Nirmal4G) [inline eng/Versions.props:34]: See if we have `JPath`/`JsonPath` similar to XML's `XPath` in `System.Text.Json`, we could have a nice looking code...

```suggestion
    <DotNetCliVersion>$([System.Text.Json.JsonPath]::GetValueFromFile('global.json', '$.tools.dotnet(.version)?'))</DotNetCliVersion>
```

OR

```suggestion
    <DotNetCliVersion>$([System.Text.Json.JsonPath]::GetValueFromFile('global.json', '$.sdk(.version)?'))</DotNetCliVersion>
```

**Pull request merged**